### PR TITLE
Use caching for inventory opening when possible

### PR DESCRIPTION
**Pull request by DoctorMacc**
When possible, this prevents the inventory block from appearing on top of your head.

### CODE REVIEW

**Comment by LegacyGamerHD** [comment]
Again no bungee

**Comment by DoctorMacc** [comment]
> Again no bungee 

I believe Bungee is handled with the same code as standalone.

File: connector/src/main/java/org/geysermc/connector/network/translators/inventory/DoubleChestInventoryTranslator.java [hunk 1]
```diff
@@ -40,60 +40,98 @@
...
+            // Reset position to player because this is the wrong block
+            position = session.getPlayerEntity().getPosition().toInt().add(Vector3i.UP);
+            pairPosition = position.add(Vector3i.UNIT_X);
+        }
 
         UpdateBlockPacket blockPacket = new UpdateBlockPacket();
         blockPacket.setDataLayer(0);
         blockPacket.setBlockPosition(position);
-        blockPacket.setRuntimeId(blockId);
+        blockPacket.setRuntimeId(inventory.isCreatingNewBlock() ? BlockTranslator.getBedrockBlockId(BlockTranslator.getJavaBlockState(blockId)) : newBlockId);
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/inventory/DoubleChestInventoryTranslator.java:117]: Your getting the blockstate from the bedrock id to turn it back into the bedrock id, couldn't this be simplified to become:
```java
blockPacket.setRuntimeId(inventory.isCreatingNewBlock() ? blockId) : newBlockId);
```

File: connector/src/main/java/org/geysermc/connector/network/translators/inventory/DoubleChestInventoryTranslator.java [hunk 2]
```diff
@@ -40,60 +40,98 @@
...
+        // Only needed when making a new block
+        if (inventory.isCreatingNewBlock()) {
+            sendBlockEntityDataPacket(session, position, pairPosition, inventory);
+        }
 
         blockPacket = new UpdateBlockPacket();
         blockPacket.setDataLayer(0);
         blockPacket.setBlockPosition(pairPosition);
-        blockPacket.setRuntimeId(blockId);
+        blockPacket.setRuntimeId(BlockTranslator.getBedrockBlockId(BlockTranslator.getJavaBlockState(otherPairBlockId)));
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/inventory/DoubleChestInventoryTranslator.java:129]: Same as above

File: connector/src/main/java/org/geysermc/connector/network/translators/inventory/holder/BlockInventoryHolder.java [hunk 3]
```diff
@@ -42,30 +43,64 @@
...
+            // Reset position to player because this is the wrong block
+            position = session.getPlayerEntity().getPosition().toInt();
+            position = position.add(Vector3i.UP);
+        }
+
         UpdateBlockPacket blockPacket = new UpdateBlockPacket();
         blockPacket.setDataLayer(0);
         blockPacket.setBlockPosition(position);
-        blockPacket.setRuntimeId(blockId);
+        blockPacket.setRuntimeId(inventory.isCreatingNewBlock() ? BlockTranslator.getBedrockBlockId(BlockTranslator.getJavaBlockState(blockId)) : newBlockId);
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/inventory/holder/BlockInventoryHolder.java:88]: Same as above

**Comment by DoctorMacc** [comment]
Closing since we likely will need to redo the inventory system anyway for 1.16

**Pull request closed without merging**